### PR TITLE
ARROW-3214: [C++] Disable insecure warnings in MinGW build

### PR DESCRIPTION
**Pull request by kou**
It's already disabled in MSVC build.

### CODE REVIEW

**Comment by kou** [comment]
You can see the important changes easily by disabling whitespace changes by https://github.com/apache/arrow/pull/2539/files?w=1 .

**Review by pitrou — CHANGES_REQUESTED** [review]
LGTM, just one nit below.

File: cpp/cmake_modules/SetupCxxFlags.cmake [hunk 1]
```diff
@@ -25,46 +25,48 @@ CHECK_CXX_COMPILER_FLAG("-maltivec" CXX_SUPPORTS_ALTIVEC)
...
+		      CMAKE_C_FLAGS_MINSIZEREL CMAKE_C_FLAGS_RELWITHDEBINFO)
+	string(REPLACE "/MD" "-MT" ${c_flag} "${${c_flag}}")
+      endforeach()
+    endif()
 
-  # Support large object code
-  set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} /bigobj")
+    # Support large object code
+    set(CXX_COMMON_FLAGS "${CXX_COMMON_FLAGS} /bigobj")
+  endif()
```
[the remark this answers is not in the document]
Comment (kou) [reply cpp/cmake_modules/SetupCxxFlags.cmake]: That makes sense.
I've added.

**Review by xhochy — APPROVED** [review]
+1, LGTM

**Pull request closed without merging**